packages/ra-data-graphql/src/index.js

```diff
@@ -77,6 +78,17 @@ export default async options => {
 
     const buildQuery = buildQueryFactory(introspectionResults, otherOptions);
 
+    const parseError = ({ networkError, graphQLErrors }) => {
+        const message =
+            get(graphQLErrors, '[0].message') ||
+            get(networkError, 'result.errors[0].message') ||
+            get(networkError, 'message');
+        const status =
```

Comment:
Isn't this part highly dependent on the server implementation ? I'm not sure we should handle errors at all as they are many possible implementations for errors. Wouldn't it be better to add a section about error handling in the documentation?

Reply:
Oh my, just realised the `params` attribute is actually the error object in the `AUTH_ERROR` case. My bad! So we could specify that in the docs but it looks obvious now haha.

Regarding your comment, I believe this is the standard Apollo error formatting, i.e. all errors behave that way regardless of server implementations. The goal here was to reformat the error object so it would work like the other data providers.

I guess I can close the PR as all this code can be used within the auth provider implementation.

Reply:
> Oh my, just realised the params attribute is actually the error object in the AUTH_ERROR case. My bad! So we could specify that in the docs but it looks obvious now haha.

A documentation PR to clarify this point would be greatly appreciated. It is not so obvious in our current documentation ;)

> all errors behave that way regardless of server implementations

Maybe true for the `graphQLErrors`. Not so sure about the `networkError` though.